src/components/NavigationMenu/useNavigationMenu.ts

```diff
@@ -0,0 +1,41 @@
+import { useCallback, useMemo, useState } from 'react';
+
+import { authAPI } from '../../api/auth';
+
+export enum RoutesList {
+  all = '*',
+  home = '/',
+  leaderboard = 'leaderboard',
+  library = '/library',
+  logout = '/logout',
+  login = '/login',
+  play = '/play',
+  profile = '/profile',
+  register = '/register',
+}
+
+export const useNavigationMenu = () => {
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
+  const toggleMenu = () => setIsMenuOpen((prevState) => !prevState);
```

Comment:
❗️
Если из хука что-то экспортим, то мемоизируем
```suggestion
  const toggleMenu = useCallback(() => setIsMenuOpen(!isMenuOpen), [isMenuOpen]);
```

Reply:
Добавил